core/dns.go

```diff
@@ -191,20 +166,24 @@ func (dnsResolver *DNSResolver) LookupCAA(domain string, alias bool) ([]*dns.CAA
 	m := new(dns.Msg)
 	m.SetQuestion(dns.Fqdn(domain), dns.TypeCAA)
 
-	r, _, err := dnsResolver.LookupDNSSEC(m)
+	r, _, err := dnsResolver.ExchangeOne(m)
 	if err != nil {
 		return nil, err
 	}
 
 	var CAAs []*dns.CAA
-	for _, answer := range r.Answer {
-		if answer.Header().Rrtype == dns.TypeCAA {
-			caaR, ok := answer.(*dns.CAA)
-			if !ok {
-				err = errors.New("Badly formatted record")
-				return nil, err
+	// XXX: On resolver validation failure, or other server failures, return empty
+	//      set and no error.
+	if r.Rcode != dns.RcodeServerFailure {
+		for _, answer := range r.Answer {
+			if answer.Header().Rrtype == dns.TypeCAA {
+				caaR, ok := answer.(*dns.CAA)
```

Comment:
OK, yeah, since you do the checked cast here, can you please change the unchecked cast(s) above? Thanks!